src/Controller/ListPageRssController.php

```diff
@@ -0,0 +1,188 @@
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Cache\CacheableResponse;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Theme\ThemeManagerInterface;
+use Drupal\node\NodeInterface;
+use Drupal\oe_list_pages\ListPageRssBuildAlterEvent;
+use Drupal\oe_list_pages\ListPageEvents;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Controller for the List pages RSS feed routes.
+ */
+class ListPageRssController extends ControllerBase {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The event dispatcher.
+   *
+   * @var \Symfony\Component\EventDispatcher\EventDispatcherInterface
+   */
+  protected $eventDispatcher;
+
+  /**
+   * The theme manager.
+   *
+   * @var \Drupal\Core\Theme\ThemeManagerInterface
+   */
+  protected $themeManager;
+
+  /**
+   * ListPageRssController constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Symfony\Component\EventDispatcher\EventDispatcherInterface $event_dispatcher
+   *   The event dispatcher.
+   * @param \Drupal\Core\Theme\ThemeManagerInterface $theme_manager
+   *   The theme manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, EventDispatcherInterface $event_dispatcher, ThemeManagerInterface $theme_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->eventDispatcher = $event_dispatcher;
+    $this->themeManager = $theme_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): ListPageRssController {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('event_dispatcher'),
+      $container->get('theme.manager')
+    );
+  }
+
+  /**
+   * Render the RSS feed of a list page result.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The list page node.
+   *
+   * @return string
+   *   The rendered RSS feed.
+   */
+  public function build(NodeInterface $node): Response {
+    // Build the render array for the RSS feed.
+    $default_title = $node->getTitle() . ' - RSS';
+    $default_link = $node->toUrl('canonical', ['absolute' => TRUE])->toString(TRUE);
+    $build = [
+      '#theme' => 'oe_list_pages_rss',
+      '#title' => $default_title,
+      '#link' => $default_link->getGeneratedUrl(),
+      '#description' => '',
+      '#language' => $node->language()->getId(),
+      '#copyright' => $this->getCopyright(),
+      '#image_url' => $this->getDefaultImageUrl(),
```

Comment:
Don;t create separate variables for image_url, image_title, image_link. Just have it be an array with 3 keys.